ApiTests/Tests/Pages/DynamicControlsPage.cs

```diff
@@ -0,0 +1,45 @@
+﻿using ApiTests.Tests.UI;
+using OpenQA.Selenium;
+using OpenQA.Selenium.Support.UI;
+using SeleniumExtras.WaitHelpers;
+using TestProject.Utils;
+
+namespace TestProject.Tests.Pages
+{
+    internal class DynamicControlsPage : BaseTest
```

Comment:
Page shouldn't be inherited from BaseTest, they have no common logic between them
You can create a BasePage class if you have a common page logic (such as checking if the page is open or not)